forward_engineering/utils/general.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @param index {number}
+ * @param amountOfStatements {number}
+ * @param lastIndexOfActivatedStatement {number}
+ * @param delimiter {string}
+ * @param delimiterForLastActivatedStatement {string}
+ * @return {string}
+ * */
+const getDelimiterForJoiningStatementsBasedOnActivation = ({
+                                                               index,
+                                                               amountOfStatements,
+                                                               lastIndexOfActivatedStatement,
+                                                               delimiter = ',\n',
+                                                               delimiterForLastActivatedStatement = '\n',
+                                                           }) => {
+    if (index === amountOfStatements - 1) {
+        return '';
+    }
+    if (lastIndexOfActivatedStatement === -1) {
+        return delimiter;
+    }
+    if (index === lastIndexOfActivatedStatement) {
+        return delimiterForLastActivatedStatement;
+    }
+    return delimiter;
+}
+
+/**
+ * @param statementDtos {Array<{
+ *     statement: string,
+ *     isActivated: boolean,
+ * }>}
+ * @return {number}
+ */
+const getLastIndexOfActivatedStatement = (statementDtos) => {
+     for (let i = statementDtos.length - 1; i >= 0; i--) {
+         const statementDto = statementDtos[i] || {};
+         if (statementDto.isActivated) {
+             return i;
+         }
+     }
+     return -1;
+}
+
+
+/**
+ * @param statementDtos {Array<{
+ *     statement: string,
+ *     isActivated: boolean,
+ * }>}
+ * @param delimiter {string | undefined}
+ * @param delimiterForLastActivatedStatement {string | undefined}
+ * @return {string}
+ * */
+const joinActivatedAndDeactivatedStatements = ({
+                                                   statementDtos,
+                                                   delimiter = ',\n',
+                                                   delimiterForLastActivatedStatement = '\n',
```

Comment:
if you are going to use these params only within `getDelimiterForJoiningStatementsBasedOnActivation` [function](https://github.com/hackolade/BigQuery/pull/45/files#diff-e31abefe4112fbf65021241569c69f0adf6a22ad5a448437d552790ac68272beR13-R14), then I can't see the reason providing a default values here

Reply:
Spreading default value declarations throughout several functions is a bad design + hard to read